### PR TITLE
Metric comparison

### PR DESCRIPTION
**Pull request by chaws**
Add option for metric based comparison for builds/projects. It's using the same template from test comparison. It shows the mean, the standard deviation between parenthesis and the number of measurements when hovering on the table cell.

Screenshot from 2019-06-28 17-21-56

Screenshot from 2019-06-28 17-24-21

### CODE REVIEW

File: squad/core/comparison.py [hunk 1]
```diff
@@ -107,18 +78,96 @@ def diff(self):
...
         return self.__diff__
 
+
+class MetricComparison(BaseComparison):
+
+    def __get_build__(self, build_id):
+        for build in self.builds:
+            if build.id == build_id:
+                return build
+        return None
```
Comment (terceiro) [inline squad/core/comparison.py]: you probably want to use a `dict` here

File: squad/frontend/comparison.py [hunk 2]
```diff
@@ -2,16 +2,22 @@
 from django.core.paginator import Paginator
 
 from squad.core.models import Project
-from squad.core.comparison import TestComparison
+from squad.core.comparison import TestComparison, MetricComparison
+
+
+def __get_comparison_class(comparison_type):
+    return MetricComparison if 'metric' == comparison_type else TestComparison
```
Comment (terceiro) [inline squad/frontend/comparison.py]: I think a regular multi-line if is easier to read

File: squad/frontend/templates/squad/compare_projects.jinja2 [hunk 3]
```diff
@@ -29,6 +29,16 @@
...
+  <div class="form-group">
+    <label>{{ _('compare by') }}</label>
+    <div>
+      <label for="test-comparison">{{ _('tests') }}</label>
+      <input id="test-comparison" name="comparison_type" type="radio" value="test" checked="checked"/>
+      &nbsp;{{ _('or') }}&nbsp;
+      <label for="metric-comparison">{{ _('metrics') }}</label>
+      <input id="metric-comparison" name="comparison_type" type="radio" value="metric" />
+    </div>
+  </div>
```
Comment (terceiro) [inline squad/frontend/templates/squad/compare_projects.jinja2]: this does not work for internationalization; you should always use full sentences instead, e.g.
[ ] compare by tests
[ ] compare by metrics

when you break up the sentences, you assume that all languages have the same flow structure as English, and that is not the case.

Comment (chaws) [reply squad/frontend/templates/squad/compare_projects.jinja2]: That's true, I didn't consider that into the equation. I've added a group label as "Comparison type", then added the options as you suggested. Looks much better indeed.

**Comment by chaws** [comment]
@terceiro , just pushed new updates with your reviews and with mean and stdev being calculated in python using 'statistics' module. It's supposed to be supported in Python 3+.

File: squad/frontend/templates/squad/compare_builds.jinja2 [hunk 4]
```diff
@@ -36,10 +36,20 @@
...
+		  <div class="form-group">
+			<label>{{ _('Comparison type') }}</label>
+			<div>
+			  <input id="test-comparison" name="comparison_type" type="radio" value="test" checked="checked"/>
+			  <label for="test-comparison">{{ _('compare by tests') }}</label>
+			  <br />
+			  <input id="metric-comparison" name="comparison_type" type="radio" value="metric" />
+			  <label for="metric-comparison">{{ _('compare by metrics') }}</label>
+			</div>
+		  </div>
```
Comment (terceiro) [inline squad/frontend/templates/squad/compare_builds.jinja2]: the indentation here is wrong

File: squad/frontend/templates/squad/compare_builds.jinja2 [hunk 5]
```diff
@@ -36,10 +36,20 @@
...
           <div class="form-group">
             <input type="hidden" name="project" value="{{ project.full_name }}" />
             <input onclick="this.form.submit()" value="{{ _('Compare') }}" type="button" class="btn btn-default" />
-            or
+            {{ _('or') }}
```
Comment (terceiro) [inline squad/frontend/templates/squad/compare_builds.jinja2]: I think we should drop this "or". everybody knows what to do when there is a sequence of buttions

File: squad/frontend/templates/squad/compare_builds.jinja2 (hunk 4 above)
Comment (chaws) [reply squad/frontend/templates/squad/compare_builds.jinja2]: hm, I my vim got confused with tabs. Fixed that, thanks for spotting that.

File: squad/frontend/templates/squad/compare_builds.jinja2 (hunk 5 above)
Comment (chaws) [reply squad/frontend/templates/squad/compare_builds.jinja2]: agreed.

**Comment by chaws** [comment]
Nitpicks are always welcome :)

**Comment by chaws** [comment]
Closed #201

**Pull request merged**